### PR TITLE
Apply RFC to IPv6 to string [10252]

### PR DESCRIPTION
**Pull request by jparisu**
RFC recommendation: https://tools.ietf.org/html/rfc5952
Related to #1653

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3528/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/718/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3136/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3560/)

File: src/cpp/utils/IPLocator.cpp [hunk 1]
```diff
@@ -454,31 +454,90 @@ bool IPLocator::hasIPv6(
...
+     *  4.2.1   Shorten as much as possible (do not leave 0 blocks at sides of ::)
+     *  4.2.2   No collapse size 1 blocks
+     *  4.2.3   Collapse the largest block (or the first one in case of tie)
+     *  4.3     Lowercase
+     */
+
+    int max_block_index = 0;
+    int max_block_size = 0;
+    int actual_block_index = 0;
+    int actual_block_size = 0;
```
Comment (EduPonz) [inline src/cpp/utils/IPLocator.cpp:468]: These could be unsigned, and I guess they could be smaller.

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]
LGTM

**Review by RFRIEDM — COMMENTED** [review]
LGTM

**Pull request merged**